### DISCUSSION TITLE
App Extensions vs Mono Repo??

### DISCUSSION THREAD

**Discussion by fabian-murillo** [quasarframework/quasar#9345] [Quasar App Extensions] [answered]
Recently I've been researching app extensions, I've been considering using them to create a common component library. Through my research, I had various people recommend I checkout yarn workspaces, and so I did. Now I'm left with a question...

Why would I want to create a common component library and publish it on npm as a quasar app extension vs setting up a monorepo with shared components/pages/etc. through yarn workspaces?

Looking for pros/cons & opinions on both

Additional Information:
I work on a 8 person team (4 front end devs), we have several quasar projects, each in separate GitHub repositories. Currently our process has been copying and pasting many of the same files over into new projects. This practice has created a lot of problems and I'm trying to find a better solution.

Thanks!

**Comment by hawkeye64** [comment, marked as answer]
@fabian-murillo 
You write app-extensions to do something. It could be UI-based or extend the CLI.
You use a workspace to share. For work, I use a workspace. 
For pleasure, I write app-extensions I feel will be useful for others.
I don't really consider them to be the same at all.

**Reply by fabian-murillo** [reply]
That makes sense, thank you :pray: .
